### PR TITLE
MINOR: Fix 3.1 upgrade notes for idempotence bug

### PR DESCRIPTION
**Pull request by hachikuji**
I accidentally pulled in the 3.2 notes into the 3.1 branch after cherry-picking https://github.com/apache/kafka/pull/11691. 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: docs/upgrade.html [hunk 1]
```diff
@@ -19,56 +19,6 @@
...
 
-<h5><a id="upgrade_320_notable" href="#upgrade_320_notable">Notable changes in 3.2.0</a></h5>
-    <ul>
-        <li>Idempotence for the producer is enabled by default. In 3.0.0 and 3.1.0, a bug prevented this default from being applied,
-            which meant that idempotence remained disabled unless the user had explicitly set <code>enable.idempotence</code> to true
-            (See <a href="https://issues.apache.org/jira/browse/KAFKA-13598">KAFKA-13598</a>for more details).
-            This issue was fixed and the default is properly applied in 3.0.1, 3.1.1, and 3.2.0.</li>
-    </ul>
-
-<h4><a id="upgrade_3_1_0" href="#upgrade_3_1_0">Upgrading to 3.1.0 from any version 0.8.x through 3.0.x</a></h4>
```
Comment (showuon) [inline docs/upgrade.html:30]: I think this section should still be kept: `Upgrading to 3.1.0 from any version 0.8.x through 3.0.x`

**Review by showuon — COMMENTED** [review]
Nice catch! There's a section is deleted accidentally. Thanks.

**Review by ijuma — APPROVED** [review]
LGTM if @showuon's comment is addressed.

File: docs/upgrade.html (hunk 1 above)
Comment (hachikuji) [reply docs/upgrade.html:30]: Thanks for catching!

**Pull request merged**